iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -493,18 +503,42 @@ private String getStatsSource() {
   }
 
   private Path getStatsPath(Table table) {
-    return new Path(table.location() + STATS + table.name() + table.currentSnapshot().snapshotId());
+    return getStatsPath(table, table.currentSnapshot().snapshotId());
   }
 
-  private void invalidateStats(Path statsPath) {
+  private Path getStatsPath(Table table, long snapshotId) {
+    return new Path(table.location() + STATS + table.name() + snapshotId);
+  }
+
+  private boolean checkAndInvalidateStats(Table tbl) {
+    Path statsPath = getStatsPath(tbl);
     try {
       FileSystem fs = statsPath.getFileSystem(conf);
       if (fs.exists(statsPath)) {
-        fs.delete(statsPath, true);
+        // Analyze table and stats updater thread
+        return fs.delete(statsPath, true);
       }
     } catch (IOException e) {
       LOG.error("Failed to invalidate stale column stats: {}", e.getMessage());
```

Comment:
should it be a `warn` or `error`? even if we failed to execute `delete` we are trying to merge the stats?

Reply:
btw, why do we need to delete, can't we simply overwrite the file with new stats?

Reply:
Puffin writes calls create instead of createoverwrite in PositionOutputStream . 

https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/puffin/PuffinWriter.java#L66 

As a result, it "throws file already exists exception". So we had to delete and write instead of overwrite.
Or fix in iceberg repo.

Reply:
> should it be a warn or error?

Changed it to warn